TidepoolKit/Model/Datum/TCGMSettingsDatum.swift

```diff
@@ -0,0 +1,240 @@
+        self.manufacturers = manufacturers
+        self.model = model
+        self.serialNumber = serialNumber
+        self.transmitterId = transmitterId
+        self.units = units
+        self.defaultAlerts = defaultAlerts
+        self.scheduledAlerts = scheduledAlerts
+        super.init(.cgmSettings, time: time)
+    }
+
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.manufacturers = try container.decodeIfPresent([String].self, forKey: .manufacturers)
+        self.model = try container.decodeIfPresent(String.self, forKey: .model)
+        self.serialNumber = try container.decodeIfPresent(String.self, forKey: .serialNumber)
+        self.transmitterId = try container.decodeIfPresent(String.self, forKey: .transmitterId)
+        self.units = try container.decodeIfPresent(Units.self, forKey: .units)
+        self.defaultAlerts = try container.decodeIfPresent(Alerts.self, forKey: .defaultAlerts)
+        self.scheduledAlerts = try container.decodeIfPresent([ScheduledAlert].self, forKey: .scheduledAlerts)
+        self.highAlertsDEPRECATED = try container.decodeIfPresent(LevelAlertsDEPRECATED.self, forKey: .highAlertsDEPRECATED)
+        self.lowAlertsDEPRECATED = try container.decodeIfPresent(LevelAlertsDEPRECATED.self, forKey: .lowAlertsDEPRECATED)
+        self.outOfRangeAlertsDEPRECATED = try container.decodeIfPresent(OutOfRangeAlertsDEPRECATED.self, forKey: .outOfRangeAlertsDEPRECATED)
+        self.rateOfChangeAlertsDEPRECATED = try container.decodeIfPresent(RateOfChangeAlertsDEPRECATED.self, forKey: .rateOfChangeAlertsDEPRECATED)
+        try super.init(.cgmSettings, from: decoder)
+    }
+
+    public override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encodeIfPresent(manufacturers, forKey: .manufacturers)
+        try container.encodeIfPresent(model, forKey: .model)
+        try container.encodeIfPresent(serialNumber, forKey: .serialNumber)
+        try container.encodeIfPresent(transmitterId, forKey: .transmitterId)
+        try container.encodeIfPresent(units, forKey: .units)
+        try container.encodeIfPresent(defaultAlerts, forKey: .defaultAlerts)
+        try container.encodeIfPresent(scheduledAlerts, forKey: .scheduledAlerts)
+        try container.encodeIfPresent(highAlertsDEPRECATED, forKey: .highAlertsDEPRECATED)
+        try container.encodeIfPresent(lowAlertsDEPRECATED, forKey: .lowAlertsDEPRECATED)
+        try container.encodeIfPresent(outOfRangeAlertsDEPRECATED, forKey: .outOfRangeAlertsDEPRECATED)
+        try container.encodeIfPresent(rateOfChangeAlertsDEPRECATED, forKey: .rateOfChangeAlertsDEPRECATED)
+        try super.encode(to: encoder)
+    }
+
+    public struct Alerts: Codable, Equatable {
+        public var enabled: Bool?
```

Comment:
Curious what this enabled refers to? Does this mean that the alerts are enabled? If so, at least with the Dexcom integration, the mapping of alerts to enabled is 1-to-1 and some alerts cannot be disabled. Or does this refer to always sound is enabled? Then the mapping of alerts to enabled is many-to-1, but again some alerts always sounds cannot be disabled (if critical alerts has been allowed by the user). The Tidepool Loop model has both an enabled and always sound enabled property for each alert.

Reply:
It was promoted from the `ScheduledAlert` struct below.